File: simplified-app/src/main/java/org/nypl/simplified/app/reader/ReaderSettingsListenerType.java
package org.nypl.simplified.app.reader;

/**
 * The type of listeners that receive settings changes.
 */

public interface ReaderSettingsListenerType
{
  /**
   * The current settings have changed.
   *
   * @param s The current settings
   */

  void onReaderSettingsChanged(
    ReaderSettingsType s);
}

File: simplified-app-shared/src/main/java/org/nypl/simplified/app/reader/ReaderSettingsListenerType.kt
package org.nypl.simplified.app.reader

import org.nypl.simplified.app.ScreenSizeInformationType
import org.nypl.simplified.profiles.controller.api.ProfilesControllerType

/**
 * The listener interface that activities hosting [ReaderSettingsDialog] must implement.
 */

interface ReaderSettingsListenerType {

  /**
   * Get a reference to the current profiles controller.
   */

  fun onReaderSettingsDialogWantsProfilesController(): ProfilesControllerType

  /**
   * Get a reference to the current screen size information.
   */

  fun onReaderSettingsDialogWantsScreenSize(): ScreenSizeInformationType
}
